src/monogame/sdl/MGP_sdl.cpp

```diff
@@ -0,0 +1,386 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+#include "csharp_MGP.h"
+
+#include "stl_common.h"
+
+#include <sdl.h>
+
+
+struct MGP_Platform
+{
+	mgbool exit = false;
+
+	std::vector<MGP_Window*> windows;
+};
+
+
+struct MGP_Window
+{
+	MGP_Platform* platform;
+
+	std::string identifier;
+
+	SDL_Window* window;
+};
+
+MGP_Platform* MGP_Platform_Create(MGGameRunBehavior* behavior)
+{
+	assert(behavior != nullptr);
+
+	SDL_Init(
+		SDL_INIT_VIDEO |
+		SDL_INIT_JOYSTICK |
+		SDL_INIT_GAMECONTROLLER |
+		SDL_INIT_HAPTIC);
+
+	SDL_DisableScreenSaver();
+
+	SDL_SetHint("SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS", "0");
+	SDL_SetHint("SDL_JOYSTICK_ALLOW_BACKGROUND_EVENTS", "1");
+
+	*behavior = MGGameRunBehavior::Synchronous;
+
+	auto platform = new MGP_Platform();
+	return platform;
+}
+
+void MGP_Platform_Destroy(MGP_Platform* platform)
+{
+	assert(platform != nullptr);
+
+	// Destroy any active windows that may have been leaked.
+	for (auto window : platform->windows)
+		MGP_Window_Destroy(window);
+
+	delete platform;
+}
+
+void MGP_Platform_BeforeInitialize(MGP_Platform* platform)
+{
+	assert(platform != nullptr);
+}
+
+bool MGP_Platform_Tick(MGP_Platform* platform)
+{
+	assert(platform != nullptr);
+
+    SDL_Event ev;
+
+    while (SDL_PollEvent(&ev) == 1)
+    {
+        switch (ev.type)
+        {
+        case SDL_QUIT:
+            platform->exit = true;
+            break;
+
+        /*
```

Comment:
Just a reminder to not leave so much comments.

Reply:
WIP...  i expect these will be restored in the next week.